### PR TITLE
Choose priority when adding a new task from cli

### LINKED ISSUE #239: Choose priority when adding a new task from cli
Describtion sais it all, sth like:

```
    todo new --priority high 'Choose priority when adding a new task'
```

### PR DESCRIPTION
**Pull request by AnubhaAgrawal**
Fixes #239 Choose priority when adding a new task from cli

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -868,3 +868,22 @@ def test_invoke_invalid_command(runner, tmpdir):
...
+        'Priority:High\n'
+    )
+
+    result = runner.invoke(cli, ['show', '1'])
+    assert 'High' in result.output
+
+
+def test_priority(runner):
+    result = runner.invoke(cli, [
+        'new', '-l', 'default', '--priority', 'Chembur', 'Event Test'
```
Comment (untitaker) [inline tests/test_cli.py]: Here you invoke `--priority` with the value `Chembur`, which is not a valid priority.

File: todoman/formatters.py [hunk 2]
```diff
@@ -82,6 +82,8 @@ def detailed(self, todo):
             extra_rows += self._columnize('Description', todo.description)
         if todo.location:
             extra_rows += self._columnize('Location', todo.location)
+        if todo.priority:
```
Comment (untitaker) [inline todoman/formatters.py]: This is not necessary (and the corresponding test as well), because the priority is already shown as part of `self.compact(todo)`

File: todoman/cli.py [hunk 3]
```diff
@@ -141,6 +141,10 @@ def validate_status(ctx=None, param=None, val=None):
 
 
 def _todo_property_options(command):
+    click.option(
+        '--priority', default='', callback=_validate_priority_param,
+        help=('Only show tasks with'
```
Comment (untitaker) [inline todoman/cli.py]: This help text is wrong, we *set* the priority, not *filter* it

File: todoman/cli.py [hunk 4]
```diff
@@ -141,6 +141,10 @@ def validate_status(ctx=None, param=None, val=None):
...
+              ' priority such as low, medium or high.'))(command)
```
Comment (hobarrera) [inline todoman/cli.py]: I don't think `(command)` should be here.

Comment (untitaker) [inline todoman/cli.py]: @hobarrera `click.option(..)` returns a decorator, and this is one way of applying it.

**Comment by AnubhaAgrawal** [comment]
I make changes according to the suggestions given by @untitaker but again test cases are failed.

File: tests/test_cli.py [hunk 5]
```diff
@@ -874,16 +874,16 @@ def test_show_priority(tmpdir, runner, create):
     create(
         'test.ics',
         'SUMMARY:harhar\n'
-        'Priority:High\n'
+        'Priority:high\n'
```
Comment (hobarrera) [inline tests/test_cli.py]: Priority here should be `1`. You should really use the `todo_factory` helper instead of `create`.

File: todoman/cli.py [hunk 6]
```diff
@@ -141,6 +141,9 @@ def validate_status(ctx=None, param=None, val=None):
 
 
 def _todo_property_options(command):
+    click.option(
+        '--priority', default='', callback=_validate_priority_param,
+        help=('Only set the tasks with priority'))(command)
```
Comment (hobarrera) [inline todoman/cli.py]: Not a very accurate help text. Use the two immediately below as a reference.

File: tests/test_cli.py [hunk 7]
```diff
@@ -868,3 +868,22 @@ def test_invoke_invalid_command(runner, tmpdir):
     assert (
         'Error: Invalid setting for [main][default_command]' in result.output
     )
+
+
+def test_show_priority(tmpdir, runner, create):
+    create(
+        'test.ics',
+        'SUMMARY:harhar\n'
+        'Priority:high\n'
```
Comment (hobarrera) [inline tests/test_cli.py]: Looks like GitHub ate my previous comment.

Priority in files is numeric; should be `1` in this case. You should really use the `todo_factory` helper in this case anyway.

**Comment by untitaker** [comment]
@hobarrera I see that you reviewed a testcase that doesn't test the new functionality.

@AnubhaAgrawal in https://github.com/pimutils/todoman/pull/242/commits/8bc088823cc72569c035cd330c3bdf06b5a6b09b you removed a testcase, but that was the more important one

File: tests/test_cli.py [hunk 8]
```diff
@@ -868,3 +868,19 @@ def test_invoke_invalid_command(runner, tmpdir):
     assert (
         'Error: Invalid setting for [main][default_command]' in result.output
     )
+
+
+def test_show_priority(runner, todo_factory, todos):
+    todo = todo_factory(summary='harhar\n', priority=1)
+
+    result = runner.invoke(cli, ['show', '1'])
+    assert '1' in result.output
```
Comment (hobarrera) [inline tests/test_cli.py]: Output should say `high`, not `1`.

Comment (AnubhaAgrawal) [inline tests/test_cli.py]: if I write output as  ```high``` in place of ```1``` then this test case become failed.

Comment (hobarrera) [inline tests/test_cli.py]: The test passes because `1` is printed as the todo's ID, not priority.
Sorry, I mixed up a bit. In this case, `!!!` is printed to detonate the high priority, not `high`.

Sorry, my bad there! 😅

File: todoman/cli.py (hunk 4 above)
Comment (hobarrera) [inline todoman/cli.py]: Ugh, ignore me. I need to spot reviewing stuff right after I get up. :(

File: tests/test_cli.py (hunk 8 above)
Comment (AnubhaAgrawal) [inline tests/test_cli.py]: Now, all test cases are passed except the one ```run_test(sort_key)```. I don't understand why it is failed.

Comment (hobarrera) [inline tests/test_cli.py]: I see those tests passing, but you have [style errors](https://travis-ci.org/pimutils/todoman/jobs/226445642#L215-L216).

You can run `tox -e flake8` to test these locally.

File: todoman/formatters.py [hunk 9]
```diff
@@ -82,7 +82,6 @@ def detailed(self, todo):
             extra_rows += self._columnize('Description', todo.description)
         if todo.location:
             extra_rows += self._columnize('Location', todo.location)
-
```
Comment (hobarrera) [inline todoman/formatters.py]: I think removing this line was a mistake, everything else looks fine by me.

Comment (AnubhaAgrawal) [inline todoman/formatters.py]: by adding this line the test cases ```test_show_priority``` and ```test_priority``` are failed.

Comment (AnubhaAgrawal) [inline todoman/formatters.py]: Now, please reply what I should do??

Comment (hobarrera) [inline todoman/formatters.py]: An empty line shouldn't make a test fail (definitely not in this context). Are you sure you're not mistaken with something else?

Comment (AnubhaAgrawal) [inline todoman/formatters.py]: Now I don't understand where is a mistake? and when I check it locally it worked as we want. but why some test cases are failed I really don't understand.

Comment (AnubhaAgrawal) [inline todoman/formatters.py]: please reply, I think I do all requested changes Now, what change I should do because when I run this manually it work accordingly as we want.

Comment (hobarrera) [inline todoman/formatters.py]: This particular change is the removal of an empty line. It does not alter any functionality, and the change is completely unrelated to this PR.

File: todoman/formatters.py [hunk 10]
```diff
@@ -82,7 +82,7 @@ def detailed(self, todo):
...
+
```
Comment (hobarrera) [inline todoman/formatters.py]: Please remove the newly-added empty spaces here. Spaces at the end of the line will make style checks fail.

File: todoman/cli.py [hunk 11]
```diff
@@ -141,6 +141,9 @@ def validate_status(ctx=None, param=None, val=None):
 
 
 def _todo_property_options(command):
+    click.option(
+        '--priority', default='', callback=_validate_priority_param,
+        help=('The priority of the task is set.'))(command)
```
Comment (hobarrera) [inline todoman/cli.py]: This is probably clearer written as `'The priority for this todo'`.

**Comment by hobarrera** [comment]
Looks good, thanks!

@untitaker comments?

**Comment by untitaker** [comment]
LGTM

**Comment by hobarrera** [comment]
Thanks, @AnubhaAgrawal! 🎉
